app/src/main/java/com/vci/vectorcamapp/imaging/presentation/components/specimen/gallery/CapturedSpecimenOverlay.kt

```diff
@@ -17,58 +18,82 @@ import androidx.compose.material3.Icon
 import com.vci.vectorcamapp.imaging.presentation.components.specimen.infocard.SpecimenInfoCard
-import com.vci.vectorcamapp.imaging.presentation.model.BoundingBoxUi
 
 @Composable
 fun CapturedSpecimenOverlay(
     specimen: Specimen,
-    boundingBoxUi: BoundingBoxUi,
+    boundingBox: BoundingBox,
     modifier: Modifier = Modifier,
     specimenBitmap: Bitmap? = null,
     onSpecimenIdCorrected: ((String) -> Unit)? = null,
     onRetakeImage: (() -> Unit)? = null,
     onSaveImageToSession: (() -> Unit)? = null,
 ) {
     val context = LocalContext.current
+    var overlaySize by remember { mutableStateOf(IntSize.Zero) }
+
+    val aspectRatio = 4f / 3f
 
     Box(
-        modifier = modifier.fillMaxSize(), contentAlignment = Alignment.BottomCenter
+        modifier = modifier.fillMaxSize(),
+        contentAlignment = Alignment.BottomCenter
     ) {
-        if (specimenBitmap != null) {
-            Image(
-                bitmap = specimenBitmap.asImageBitmap(),
-                contentDescription = specimen.id,
-                contentScale = ContentScale.Fit,
-                modifier = Modifier.fillMaxSize()
-            )
-        } else if (specimen.imageUri != Uri.EMPTY) {
-            AsyncImage(
-                model = ImageRequest.Builder(context).data(specimen.imageUri).crossfade(true)
-                    .build(),
-                contentDescription = specimen.id,
-                contentScale = ContentScale.Fit,
-                modifier = Modifier.fillMaxSize()
-            )
-        }
+        Box(
+            modifier = Modifier.fillMaxSize(),
+            contentAlignment = Alignment.Center
+        ) {
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .aspectRatio(1f / aspectRatio)
+                    .onSizeChanged { overlaySize = it }
+            ) {
+                if (specimenBitmap != null) {
+                    Image(
+                        bitmap = specimenBitmap.asImageBitmap(),
+                        contentDescription = specimen.id,
+                        contentScale = ContentScale.FillBounds,
+                        modifier = Modifier.fillMaxSize()
+                    )
+                } else if (specimen.imageUri != Uri.EMPTY) {
+                    AsyncImage(
+                        model = ImageRequest.Builder(context).data(specimen.imageUri).crossfade(true)
+                            .build(),
+                        contentDescription = specimen.id,
+                        contentScale = ContentScale.FillBounds,
+                        modifier = Modifier.fillMaxSize()
+                    )
+                }
 
-        BoundingBoxOverlay(
-            boundingBoxUi = boundingBoxUi, modifier = Modifier.fillMaxSize()
-        )
+                BoundingBoxOverlay(
+                    boundingBoxes = listOf(boundingBox),
+                    overlaySize = overlaySize,
```

Comment:
If you take a single bounding box as argument, you dont need to pass in listOf(boundingBox) which is a bit awkward given that you will only ever have one bounding box per specimen.